### Фибоначчивая система исчисления.

**Теорема:** Любое целое неотрицательное число можно представить в виде суммы *не соседних* чисел Фибоначчи.

In [5]:
F = [0, 1]
for i in range(2, 21):
    F.append(F[i - 1] + F[i - 2])
    
print(F)

[0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610, 987, 1597, 2584, 4181, 6765]


**Примеры:**

$N_1 = 47 = 34 + 13$  
$N_{1_{fib}} = 1010000000$  

$N_2 = 100 = 89 + 11 = 89 + 8 + 3$  
$N_{2_{fib}} = 100001010000$  

$N_3 = 937 = 610 + 327 = 610 + 233 + 94 = 610 + 233 + 89 + 5$  
$N_{3_{fib}} = 1010100000100000$


#### Алгоритм перевода целых чисел из десятичной системы счисления в ФСС.

$1$. Найдем максимальное число $F$ из базиса фибоначчиевой системы, которое не превосходит число $A$, и зафиксируем его номер $k$ ($F = F_k$). Искомая запись числа $A$ в фибоначчиевой системе будет содержать $k$ цифр. Сформируем “заготовку” искомого фибоначчиева числа в виде $10...0$. (нулей - $k-1$ штук).  
  
$2$. Вычислим разность $A = A - F$.  
  
$3$. Если полученное значение $A$ равно $0$, конец алгоритма, переходим к пункту $7$.  
  
$4$. Иначе, найдем максимальное число $F$ из базиса фибоначчиевой системы, которое не превосходит число $A$,  
     и зафиксируем его номер $p$ ($F = F_p$).  
       
$5$. В сформированную заготовку на $p$ -е место поставим $1$.  
  
$6$. Положим $k = p$. Переход на п. $2$.  
  
$7$. Сформированная “заготовка” является искомым числом в ФСС. Конец алгоритма.  